src/lib.rs

```diff
@@ -468,6 +471,132 @@ impl From<io::Error> for GooseError {
     }
 }
 
+#[derive(Clone, Debug, PartialEq)]
+/// A GooseAttack load test can operate in only one mode.
+pub enum GooseMode {
+    /// A mode has not yet been assigned.
+    Undefined,
+    /// A single standalone process performing a load test.
+    StandAlone,
+    /// The controlling process in a Gaggle distributed load test.
+    Manager,
+    /// One of one or more working processes in a Gaggle distributed load test.
+    Worker,
+}
+
+/// Optional default values for Goose run-time options.
+#[derive(Clone, Debug, Default)]
+pub struct GooseDefaults {
```

Comment:
Why not call this `GooseConfig` and then allow to set this config as default for an attack or as configuration per task? Just reuse the same thing. Or maybe that's the idea and it is just the name confusing me.

Reply:
I think the fundamental confusion is you're trying to modify Goose state directly, instead of using the provided functions. This was necessary prior to `0.10`, but I recommend once you upgrade that you use `set_default()` to modify Goose's configuration state.

You should not be editing `goose_attack.configuration` nor `goose_attack.defaults` directly.